### PR TITLE
Adds target happy model es24 tx slim pro 2400 tx

### PR DESCRIPTION
**Pull request by JyeSmith**
http://www.happymodel.cn/index.php/2021/08/26/happymodel-2-4g-expresslrs-es24tx-slim-pro-tx-module-for-x-lite-tango2/

This PR also sneaks in HAS_BUILT_IN_FAN.

A follow up PR will be needed to add a MinPower define for targets that can not go as low as 10mW e.g. HM Slim Pro and Namimno.

### CODE REVIEW

File: src/lib/POWERMGNT/POWERMGNT.h [hunk 1]
```diff
@@ -14,14 +14,14 @@
     defined(TARGET_R9M_TX)        || \
     defined(TARGET_TX_ES915TX)    || \
     defined(TARGET_ES900TX)
-#if defined(UNLOCK_HIGHER_POWER) || defined(TARGET_NAMIMNORC_TX)
+#if defined(UNLOCK_HIGHER_POWER) || defined(HAS_BUILT_IN_FAN)
```
Comment (cruwaller) [inline src/lib/POWERMGNT/POWERMGNT.h:17]: you should remove `HAS_BUILT_IN_FAN` and set `UNLOCK_HIGHER_POWER` for NamimnoRC targets. 
There could be a target which has a FAN but not capable for higher power :)

Comment (JyeSmith) [reply src/lib/POWERMGNT/POWERMGNT.h:17]: Good point. Will do.

**Comment by jurgelenas** [comment]
I have added these targets in https://github.com/ExpressLRS/ExpressLRS-Configurator/commit/0ef171a47b127262cae4f2578c078538653aea16

**Review by CapnBry — APPROVED** [review]

File: src/targets/happymodel_2400.ini [hunk 2]
```diff
@@ -16,6 +16,17 @@ build_flags =
 	${env:DIY_2400_TX_ESP32_SX1280_E28_via_UART.build_flags}
 	-D TARGET_HappyModel_ES24TX_2400_TX=1
 
+[env:HappyModel_ES24TX_Slim_Pro_2400_TX_via_UART]
+extends = env:DIY_2400_TX_ESP32_SX1280_E28_via_UART
+build_flags =
+	${env:DIY_2400_TX_ESP32_SX1280_E28_via_UART.build_flags}
+	-D TARGET_HappyModel_ES24TX_Slim_Pro_2400_TX=1
+	-D TX_DEVICE_NAME='"HM Slim Pro"'
```
[the remark this answers is not in the document]
Comment (cruwaller) [reply src/targets/happymodel_2400.ini]: Agreed. 
That's the way how it was (if not anymore) at some point...

**Pull request merged**